Update roletag

labcolab opened the pull request.

gokcedilek committed:
intermediate rolelist

gokcedilek committed:
create project with choosing roles

Harkunwar committed:
make role buttons clickable (#11)

* make role buttons clickable

* change to id

gokcedilek committed:
save project with role tag updates

labcolab commented:
@Harkunwar did the renaming to createPost we talked about, did a little change to save the keys of selectedRoles instead of the whole object to firestore, and tested saving a project with our updated role tag logic. should we merge this as it is since the PR includes a lot of changes, and i'll tackle the styling in a separate PR?

gokcedilek committed:
clear chosen roles after submitting form

Harkunwar commented on src/components/CreatePost/CreatePost.tsx:
```diff
@@ -1,50 +1,75 @@
-    roles: [],
   };
+
+  const validationSchema = Yup.object({
```
Since this object does not use any props/state variables, this should be declared above the function.

Harkunwar commented on src/components/CreatePost/CreatePost.tsx:
```diff
@@ -1,50 +1,75 @@
+      validationSchema={validationSchema}
       onSubmit={async (values, { resetForm }) => {
         try {
+          const savedRoles = Object.keys(selectedRoles).filter(
```
Thanks for fixing this.

Harkunwar commented:
> @Harkunwar did the renaming to createPost we talked about, did a little change to save the keys of selectedRoles instead of the whole object to firestore, and tested saving a project with our updated role tag logic. should we merge this as it is since the PR includes a lot of changes, and i'll tackle the styling in a separate PR?

Sure, that would be good.

Harkunwar commented on src/components/RoleTag/RoleTag.styles.tsx:
```diff
@@ -0,0 +1,7 @@
+import styled from '@emotion/styled';
+import { Tag } from '@chakra-ui/react';
+
+export const StyledTag = styled(Tag)`
```
Shouldn't this style be defined in `RoleList` since these margins are specific to the RoleList requirement? We can use something like StyledRoleTag.

gokcedilek committed:
tag spacing not working

labcolab commented:
@Harkunwar thanks for the review! i did the first change you suggested. for the second one, i changed roletag to just a tag, not styled, and added styling to rolelist to style our custom role tag components. however, with the way i did it the margin between the role tags do not change as i change the styling, so it didn't work like this. based on [here](https://emotion.sh/docs/styled#styling-any-component) do you think our role tag need a className prop to style it?

Harkunwar commented:
> @Harkunwar thanks for the review! i did the first change you suggested. for the second one, i changed roletag to just a tag, not styled, and added styling to rolelist to style our custom role tag components. however, with the way i did it the margin between the role tags do not change as i change the styling, so it didn't work like this. based on [here](https://emotion.sh/docs/styled#styling-any-component) do you think our role tag need a className prop to style it?

Yes! We can try that.

gokcedilek committed:
move roletag margins to rolelist

labcolab commented:
> > @Harkunwar thanks for the review! i did the first change you suggested. for the second one, i changed roletag to just a tag, not styled, and added styling to rolelist to style our custom role tag components. however, with the way i did it the margin between the role tags do not change as i change the styling, so it didn't work like this. based on [here](https://emotion.sh/docs/styled#styling-any-component) do you think our role tag need a className prop to style it?
> 
> Yes! We can try that.

wonderful, that really worked! merging this now & deleting the update_roletag branch:)
